project/Build.scala

```diff
@@ -830,6 +830,22 @@ object Build {
         baseDirectory.value.getParentFile.getParentFile / "shared/src/main/scala",
       unmanagedSourceDirectories in Test +=
         baseDirectory.value.getParentFile.getParentFile / "shared/src/test/scala",
+
+      /* The Scala 3 compiler includes this project by source. Therefore, we
+       * test that we can compile it using Scala 3, with the compiler options
+       * that are used when building the Scala 3 compiler.
+       *
+       * We do not include `-Yexplicit-nulls` although it is used in Scala 3,
+       * because we cannot cross-compile code that way. Instead, the build of
+       * Scala 3 adds an `import scala.language.unsafeNulls` in all the IR
```

Comment:
Should we do this in this repo instead in the source files (and provide a shim for Scala 2)?

Reply:
It's very tricky to provide shims for language imports. In Scala 3, they must be lexically verbatim (then later checked for semantic equivalence). That means we really have to shim that very shape (and not an alias, for example), which means that we leak a `scala.language.unsafeNulls` in our Scala 2 artifacts. Even if package private, they will exist, and potentially shadow ones used by other libraries for the same reason.